### PR TITLE
fix(graphql): fix ESM build

### LINKED ISSUE #1317: [@rollup/plugin-graphql@2.0.0] Fails to build in ESM project
- Rollup Plugin Name: @rollup/plugin-graphql
- Rollup Plugin Version: 2.0.0
- Rollup Version: 3.1.0
- Operating System (or Browser): MacOS 12.6
- Node Version: v16.x.x
- Link to reproduction _(⚠️ read below)_: https://github.com/jacekradko/rollup-plugin-graphql-esm 

Sorry for a custom repository for the reproduction of the defect, but I wasn't able to get REPL.IT to install correct version of node required for reproduction. 

**important** The reproduction project is pure ESM

### Expected Behavior

The reproduction project is an ESM module which contains a single JS file that imports a `.gql` (grapqhl) query file. The `rollup` build should successfully create a output bundle

### Actual Behavior

```bash
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/development/rollup-plugin-graphql-esm/node_modules/graphql-tag/loader' imported from /development/rollup-plugin-graphql-esm/node_modules/@rollup/plugin-graphql/dist/es/index.js
Did you mean to import graphql-tag/loader.js?
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:418:11)
    at moduleResolve (node:internal/modules/esm/resolve:983:10)
    at defaultResolve (node:internal/modules/esm/resolve:1080:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36)
```

### Additional Information

This happens both using `npm` and `yarn`. 

We were able to address this issue locally by creating a `yarn` patch for the `@rollup/plugin-graphql` . Here is the `patch` that fixes the issue: https://github.com/jacekradko/rollup-plugin-graphql-esm/pull/1

### PR DESCRIPTION
**Pull request by lukastaegert**
Use file extensions for imports

## Rollup Plugin Name: `graphql`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:
- resolves #1317

### Description
The graphql plugin was missing a file extension in its external import for proper Node ESM semantics. I added this as well as a test that actually uses the ESM bundle to validate it works now.

### CODE REVIEW

**Comment by poskadesign** [comment]
Hey! Is it possible to get this merged soon? It's a show-stopper for one of my projects.

**Comment by shellscape** [comment]
Thanks for the ping. Working on this presently.

**Review by shellscape — APPROVED** [review]
we like to have two maintainer approvals before merging. I've requested review, if no one responds in 2 days, we'll merge as is.

If your org is blocked in the meantime, you can export the graphql plugin code from the PR branch and use that locally as a stop gap.

File: packages/graphql/rollup.config.mjs
Comment (Andarist) [inline packages/graphql/rollup.config.mjs:10]: I would probably go with smth like this to be future-proof:
```suggestion
    external: id => !id.startsWith('.')
```

**Review by Kocal — APPROVED** [review]

**Pull request merged**
